terraform-modules/github-runner/startup-script.sh

```diff
@@ -0,0 +1,127 @@
+RUNNER_NAME=$(curl "http://metadata.google.internal/computeMetadata/v1/instance/name" -H "Metadata-Flavor: Google")
+
+if [[ "$ROLE_ID_PATH" == gs* ]]; then
+    gsutil cp "$ROLE_ID_PATH" "$HOME/vault-agent/role-id"
+    ROLE_ID_PATH="$HOME/vault-agent/role-id"
+fi
+
+if [[ "$SECRET_ID_PATH" == gs* ]]; then
+    gsutil cp "$SECRET_ID_PATH" "$HOME/vault-agent/secret-id"
+    SECRET_ID_PATH="$HOME/vault-agent/secret-id"
+fi
+
+# Configure users
+sudo groupadd -f docker
+id -u "$ACTIONS_USER" &>/dev/null || sudo useradd -m "$ACTIONS_USER" --groups docker
+newgrp docker
+
+# Set up environment variables for actions user
+rm -f "/home/$ACTIONS_USER/.bash_profile"
+cat <<EOF >>"/home/$ACTIONS_USER/.bash_profile"
+export VAULT_ADDR="$VAULT_ADDR"
+EOF
+
+# Use gcloud as docker credential helper
+sudo -u "$ACTIONS_USER" bash -c 'gcloud auth configure-docker --quiet'
+sudo ln -f -s /snap/google-cloud-sdk/current/bin/docker-credential-gcloud /usr/local/bin
+
+# Install software
+sudo apt-get update
+sudo apt-get -y install \
+    apt-transport-https \
+    ca-certificates \
+    gnupg-agent \
+    software-properties-common \
+    jq
+
+curl -fsSL https://download.docker.com/linux/debian/gpg | sudo apt-key add -
+sudo add-apt-repository \
+    "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) \
+    stable"
+
+curl -fsSL https://apt.releases.hashicorp.com/gpg | sudo apt-key add -
+sudo apt-add-repository \
+    "deb [arch=amd64] https://apt.releases.hashicorp.com \
+    $(lsb_release -cs) \
+    main"
+
+sudo apt-get update
+sudo apt-get -y install \
+    docker-ce \
+    docker-ce-cli \
+    containerd.io \
+    vault
+
+# Configure Vault agent
+mkdir -p "$HOME/vault-agent"
+rm -f "$HOME/vault-agent/vault-agent.hcl"
+cat <<EOF >>"$HOME/vault-agent/vault-agent.hcl"
+pid_file = "$HOME/vault-agent/pidfile"
+
+vault {
+    address = "$VAULT_ADDR"
+}
+
+auto_auth {
+    method "approle" {
+        config = {
+            role_id_file_path = "$ROLE_ID_PATH"
+            secret_id_file_path = "$SECRET_ID_PATH"
+            remove_secret_id_file_after_reading = false
+        }
+    }
+    sink "file" {
+        config = {
+            path = "/home/$ACTIONS_USER/.vault-token"
+            mode = 0660
+        }
+    }
+}
+EOF
+chmod 600 $HOME/vault-agent/*
+rm -f "$HOME/vault-agent.log"
+nohup vault agent -config="$HOME/vault-agent/vault-agent.hcl" &>"$HOME/vault-agent.log" &
+echo "Vault agent logs available in $HOME/vault-agent.log, sleeping to let it come online..."
+sleep 10s
+chown "$ACTIONS_USER" "/home/$ACTIONS_USER/.vault-token"
+
+# Configure auto-restart
+echo "0 3 * * * /sbin/shutdown -r now" | crontab -
+
+# Runner config
+mkdir -p runner
+VAULT_TOKEN=$(</home/$ACTIONS_USER/.vault-token)
+GITHUB_PAT=$(VAULT_TOKEN="$VAULT_TOKEN" vault read -address="$VAULT_ADDR" "$GITHUB_PAT_PATH" -format=json | jq -r '.data.token')
```

Comment:
Don't have to quote .data.token but OK.

Reply:
Fair enough--suppose since those are static there isn't a chance of variable substitution weirdness creeping in.